spring-cloud-deployer-local/src/main/java/org/springframework/cloud/deployer/spi/local/LocalAppDeployer.java

```diff
@@ -197,26 +204,33 @@ public void shutdown() throws Exception {
 			undeploy(deploymentId);
 		}
 	}
-	
-	protected String buildRemoteDebugInstruction(Map<String, String> deploymentProperties, String deploymentId,
-			int instanceIndex, int port) {
-		String ds = deploymentProperties.getOrDefault(LocalDeployerProperties.DEBUG_SUSPEND, "y");
-		StringBuilder debugCommandBuilder = new StringBuilder();
-		String debugCommand;
-		logger.warn("Deploying app with deploymentId {}, instance {}. Remote debugging is enabled on port {}.",
-				deploymentId, instanceIndex, port);
-		debugCommandBuilder.append("-agentlib:jdwp=transport=dt_socket,server=y,suspend=");
-		debugCommandBuilder.append(ds.trim());
-		debugCommandBuilder.append(",address=");
-		debugCommandBuilder.append(port);
-		debugCommand = debugCommandBuilder.toString();
-		logger.debug("Deploying app with deploymentId {}, instance {}.  Debug Command = [{}]", debugCommand);
-		if (ds.equals("y")) {
-			logger.warn("Deploying app with deploymentId {}.  Application Startup will be suspended until remote "
-					+ "debugging session is established.");
+
+	private Path createWorkingDir(String deploymentId, Path deploymentGroupDir) throws IOException {
+		Path workDir = Files
+				.createDirectory(Paths.get(deploymentGroupDir.toFile().getAbsolutePath(), deploymentId));
+		if (getLocalDeployerProperties().isDeleteFilesOnExit()) {
+			workDir.toFile().deleteOnExit();
 		}
+		return workDir;
+	}
+
+	private Path createLogDir(String group) throws IOException {
+		Path deploymentGroupDir = Paths.get(logPathRoot.toFile().getAbsolutePath(),
+				group + "-" + System.currentTimeMillis());
+		if (!Files.exists(deploymentGroupDir)) {
+			Files.createDirectory(deploymentGroupDir);
+//			deploymentGroupDir.toFile().deleteOnExit();
+		}
+		return deploymentGroupDir;
+	}
+
+	private void validateStatus(String deploymentId) {
+		DeploymentState state = status(deploymentId).getState();
 
-		return debugCommand;
+		if (state != DeploymentState.unknown) {
+			throw new IllegalStateException(String.format("App with deploymentId [%s] is already deployed with state [%s]",
+					deploymentId, state));
+		}
```

Comment:
Should this just be `Assert.state(DeploymentState.unknown, String.format("App.."))`?

Reply:
I like it.